File: app/src/main/java/com/example/gossipwars/ui/actions/NegotiateListAdapter.java
package com.example.gossipwars.ui.actions;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.gossipwars.R;
import com.example.gossipwars.logic.proposals.ArmyRequest;
import com.google.android.material.switchmaterial.SwitchMaterial;

import java.util.ArrayList;

public class NegotiateListAdapter extends
        RecyclerView.Adapter<NegotiateListAdapter.NegotiateViewHolder> {
    private VoteNegotiateDialog context;
    private final ArrayList<ArmyRequest> requestsList;
    private final LayoutInflater mInflater;

    class NegotiateViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public final TextView propStatement;
        public final SwitchMaterial propVote;
        final NegotiateListAdapter mAdapter;

        public NegotiateViewHolder(View itemView, NegotiateListAdapter adapter) {
            super(itemView);
            propStatement = itemView.findViewById(R.id.proposalStatement);
            propVote = itemView.findViewById(R.id.proposalSwitch);
            this.mAdapter = adapter;
            itemView.setOnClickListener(this);
            propVote.setOnCheckedChangeListener((compoundButton, isChecked) -> {
                context.sendVote(requestsList.get(getLayoutPosition()), isChecked);
            });
        }

        @Override
        public void onClick(View view) {
            int mPosition = getLayoutPosition();
            ArmyRequest element = requestsList.get(mPosition);
            mAdapter.notifyDataSetChanged();
        }
    }

    public NegotiateListAdapter(VoteNegotiateDialog context, ArrayList<ArmyRequest> requestsList) {
        this.context = context;
        mInflater = LayoutInflater.from(context.getActivity());
        this.requestsList = requestsList;
    }

    @Override
    public NegotiateViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        // Inflate an item view.
        View mItemView = mInflater.inflate(R.layout.proposal_items, parent, false);
        return new NegotiateViewHolder(mItemView, this);
    }

    @Override
    public void onBindViewHolder(NegotiateViewHolder holder, int position) {
        ArmyRequest mCurrent = requestsList.get(position);
        holder.propStatement.setText(
                String.format("%s asks you if he can raise his army %s with %s",
                        mCurrent.getInitiator().getUsername(),
                        mCurrent.getArmyOption().toString().toLowerCase(),
                        mCurrent.getIncrease()
                ));
    }

    @Override
    public int getItemCount() {
        return requestsList.size();
    }
}

File: app/src/main/java/com/example/gossipwars/ui/actions/NegotiateListAdapter.kt
package com.example.gossipwars.ui.actions

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AccelerateDecelerateInterpolator
import android.widget.CompoundButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.gossipwars.R
import com.example.gossipwars.logic.entities.GameHelper
import com.example.gossipwars.logic.entities.NegotiateVoteContent
import com.example.gossipwars.logic.entities.Snapshots
import com.example.gossipwars.ui.actions.NegotiateListAdapter.NegotiateViewHolder
import com.google.android.material.switchmaterial.SwitchMaterial
import com.richpath.RichPath
import com.richpath.RichPathView
import com.richpathanimator.RichPathAnimator
import java.util.*

class NegotiateListAdapter(
    private val context: VoteNegotiateDialog,
    private val requestsList: ArrayList<NegotiateVoteContent>
) : RecyclerView.Adapter<NegotiateViewHolder>() {
    private val mInflater: LayoutInflater = LayoutInflater.from(context.activity)

    inner class NegotiateViewHolder(
        itemView: View,
        adapter: NegotiateListAdapter
    ) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val proposalTitle: TextView = itemView.findViewById(R.id.proposalVoteTitle)
        val proposalMap: RichPathView = itemView.findViewById(R.id.proposalVoteMap)
        val proposalContent: TextView = itemView.findViewById(R.id.proposalVoteContent)
        private val propVote: SwitchMaterial = itemView.findViewById(R.id.proposalSwitch)
        val mAdapter: NegotiateListAdapter = adapter
        override fun onClick(view: View) {
            val mPosition = layoutPosition
            val content = requestsList[mPosition]
            mAdapter.notifyDataSetChanged()
        }

        init {
            itemView.setOnClickListener(this)
            propVote.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
                context.sendVote(
                    requestsList[layoutPosition].armyRequest,
                    isChecked
                )
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NegotiateViewHolder {
        // Inflate an item view.
        val mItemView = mInflater.inflate(R.layout.proposal_items, parent, false)
        return NegotiateViewHolder(mItemView, this)
    }

    override fun onBindViewHolder(holder: NegotiateViewHolder, position: Int) {
        val mCurrent = requestsList[position]
        holder.proposalTitle.text = mCurrent.title
        holder.proposalContent.text = mCurrent.content

        if (mCurrent.targetRegion == null) {
            holder.proposalMap.visibility = View.GONE
        } else {
            holder.proposalMap.setVectorDrawable(Snapshots.getDrawableForRegion(mCurrent.targetRegion))
            if (!mCurrent.membersRegions.isNullOrEmpty()) {
                for (regionId in mCurrent.membersRegions) {
                    val region: RichPath? =
                        holder.proposalMap.findRichPathByName(GameHelper.findRegionById(regionId)?.name)
                    region?.let {
                        RichPathAnimator.animate(it)
                            .interpolator(AccelerateDecelerateInterpolator())
                            .fillColor(it.fillColor, R.color.grey)
                            .start()
                    }
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return requestsList.size
    }

}